File: scripts/publish-prompt.ts
import "../src/server/load-env";
import { LangfuseClient } from "@langfuse/client";
import { env } from "../src/server/env";
import { SYSTEM_PROMPT } from "../src/server/support-agent";

const PROMPT_VARIANTS = {
  default: SYSTEM_PROMPT,
  gentler: `${SYSTEM_PROMPT.trim()}

Tone variant:
- Be extra patient and reassuring when Dad seems uncertain.
- Use plain, familiar wording before technical terms.
- Start in-scope final answers with one brief confidence-building sentence.
- If the request is out of scope, say so directly but warmly, then offer the closest iPhone-help next step.
`
} as const;

type WorkshopPromptVariant = keyof typeof PROMPT_VARIANTS;

function resolvePromptVariant(variantName: string): {
  name: WorkshopPromptVariant;
  prompt: string;
} {
  const normalizedVariantName = variantName.trim() || "default";

  if (normalizedVariantName in PROMPT_VARIANTS) {
    const name = normalizedVariantName as WorkshopPromptVariant;
    return { name, prompt: PROMPT_VARIANTS[name] };
  }

  throw new Error(
    `Unsupported WORKSHOP_PROMPT_VARIANT "${variantName}". ` +
      `Supported variants: ${Object.keys(PROMPT_VARIANTS).join(", ")}.`
  );
}

async function main() {
  const variant = resolvePromptVariant(env.workshopPromptVariant);

  if (!env.langfusePublicKey || !env.langfuseSecretKey) {
    throw new Error("Langfuse credentials are required to publish prompts.");
  }

  const langfuse = new LangfuseClient({
    publicKey: env.langfusePublicKey,
    secretKey: env.langfuseSecretKey,
    baseUrl: env.langfuseBaseUrl
  });

  const name = env.langfusePromptName || "dad-it-support-agent";
  const label = env.langfusePromptLabel || "production";

  await langfuse.prompt.create({
    name,
    type: "text",
    prompt: variant.prompt,
    labels: [label]
  });

  console.log(`Published prompt "${name}" (${variant.name} variant) with label "${label}".`);
}

void main();

File: scripts/publish_prompt.py
from __future__ import annotations

from langfuse import get_client

from backend.agent import SYSTEM_PROMPT
from backend.settings import get_settings, is_langfuse_configured


def prompt_variants() -> dict[str, str]:
    return {
        "default": SYSTEM_PROMPT,
        "gentler": f"""{SYSTEM_PROMPT.strip()}

Tone variant:
- Be extra patient and reassuring when Dad seems uncertain.
- Use plain, familiar wording before technical terms.
- Start in-scope final answers with one brief confidence-building sentence.
- If the request is out of scope, say so directly but warmly, then offer the closest
  iPhone-help next step.
""",
    }


def resolve_prompt_variant(variant_name: str) -> tuple[str, str]:
    variants = prompt_variants()
    normalized = variant_name.strip() or "default"
    if normalized not in variants:
        supported = ", ".join(sorted(variants))
        raise ValueError(
            f'Unsupported WORKSHOP_PROMPT_VARIANT "{variant_name}". '
            f"Supported variants: {supported}."
        )
    return normalized, variants[normalized]


def main() -> None:
    settings = get_settings()
    if not is_langfuse_configured(settings):
        raise RuntimeError("Langfuse credentials are required to publish prompts.")

    variant_name, prompt = resolve_prompt_variant(settings.workshop_prompt_variant)
    langfuse = get_client()
    langfuse.create_prompt(
        name=settings.langfuse_prompt_name,
        type="text",
        prompt=prompt,
        labels=[settings.langfuse_prompt_label],
    )
    langfuse.flush()

    print(
        f'Published prompt "{settings.langfuse_prompt_name}" '
        f'({variant_name} variant) with label "{settings.langfuse_prompt_label}".'
    )


if __name__ == "__main__":
    main()
